server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -73,6 +73,29 @@ object LivyConf {
   val KINIT_FAIL_THRESHOLD =
     LivyConf.Entry("livy.server.launch.kerberos.kinit_fail_threshold", 5)
 
+  /**
+   * Recovery mode of Livy. Possible values:
+   * off: Default. Turn off recovery. Every time Livy shuts down, it stops and forgets all sessions.
+   * recovery: Livy persists session info to the state store. When Livy restarts, it recovers
+   *   previous sessions from the state store.
+   * Must set livy.server.recovery.state-store and livy.server.recovery.state-store.url to
+   * configure the state store.
+   */
+  val RECOVERY_MODE = Entry("livy.server.recovery.mode", "off")
```

Comment:
Other options use `foo.enabled` (e.g. `livy.server.access_control.enabled`), I think we should follow that here too. Since there are really only two valid values. Unless there's some other mode you're thinking of adding later?